### PR TITLE
Remove unecessary reference to ClipboardStatusNotifier

### LINKED ISSUE #99360: Apply soft-transition to devtools for ClipboardStatusNotifier change
The https://github.com/flutter/flutter/commit/c74a646b7bba1641d0b2b29f6024dc867844e94c breaks devtools

```
devtools_app/lib/src/debugger/variables.dart

170: Widget buildToolbar(                                                  'VariableSelectionControls.buildToolbar' ('Widget*
                                                                           Function(BuildContext*, Rect*, double*, Offset*,
                                                                           List<TextSelectionPoint*>*, TextSelectionDelegate*,
                                                                           ClipboardStatusNotifier*, Offset*)*') isn't a valid override of
                                                                           'MaterialTextSelectionControls.buildToolbar' ('Widget
                                                                           Function(BuildContext, Rect, double, Offset,
                                                                           List<TextSelectionPoint>, TextSelectionDelegate,
                                                                           ValueListenable<ClipboardStatus>?, Offset?)'). #invalid_override

188: canCut(delegate) ? () => handleCut(delegate, clipboardStatus) : null, Too many positional arguments: 1 expected, but 2 found.
                                                                           #extra_positional_arguments

190: ? () => handleCopy(delegate, clipboardStatus)                         Too many positional arguments: 1 expected, but 2 found.
                                                                           #extra_positional_arguments

272: if (!widget.clipboardStatus.disposed) {                               The getter 'disposed' isn't defined for the type
                                                                           'ClipboardStatusNotifier'. #undefined_getter
```
We need to do a soft transition instead

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/99360

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/text_selection.dart [hunk 1]
```diff
@@ -209,7 +209,7 @@ class _CupertinoTextSelectionControlsToolbar extends StatefulWidget {
     required this.textLineHeight,
   });
 
-  final ValueNotifier<ClipboardStatus>? clipboardStatus;
+  final ValueListenable<ClipboardStatus>? clipboardStatus;
```
Comment (justinmc) [inline packages/flutter/lib/src/cupertino/text_selection.dart:212]: Why the changes from ValueNotifier to ValueListenable?

**Review by justinmc — APPROVED** [review]
LGTM 👍

File: packages/flutter/lib/src/cupertino/text_selection.dart (hunk 1 above)
Comment (chunhtai) [reply packages/flutter/lib/src/cupertino/text_selection.dart:212]: It is not using the notifier api, the builder should not be mutating the state of ClipboardStatus anyway.
